middleware/validationCheck.js

```
@@ -0,0 +1,18 @@
+const User = require("../models/user");
+
+module.exports = async (req, res, next) => {
+  const access_token = req.header("Access-Token");
+  try {
+    const valid_user = await User.findById(access_token);
+    if (!valid_user)
+      return res.status(400).json({
+        msg: "Provided token is not valid, please provide a valid token",
+      });
```

it is okay but you can throw the error from here